src/hoverifier.ts

```diff
@@ -506,6 +509,12 @@ export function createHoverifier<C extends object>({
             eventType,
             position: Position.is(position) ? { ...position, ...resolveContext(position) } : undefined,
         })),
+
+        // Suppress emissions from other events that refer to the same position as the current one. This makes it
+        // so the overlay doesn't temporarily disappear when, e.g., clicking to pin the overlay when it's already
+        // visible due to a mouseover.
+        distinctUntilChanged((a, b) => isEqual(a.position, b.position)),
```

Comment:
What about the properties other than the position like the context? E.g. if the file or rev is different but the position the same, wouldn't this filter that out?

Reply:
Hmm, yeah, it would. The `position` tokens are of type `HoveredToken | {line: number, character: number} | undefined`. If they are `{line: number, character: number}`, then this problem would occur. But in what scenarios could that happen? I assumed that because it passed that value around, it didn't need to worry about the rest of the context.

Are you thinking this would occur if there was (for example) a GitHub diff and the user moused between the red and green parts of the diff such that the line number and character remained the same? (I think that is the scenario you feared, but I just want to check.)

If so, :+1:, this can be fixed easily by adding another check for the value thrown away from `...resolveContext(position)`.

Reply:
Actually, this is not a problem. `a.position` and `b.position` are of type:

```
position: ({
    line: number;
    character: number;
    part?: "base" | "head" | null | undefined;
} & C) | undefined
```

The `C` is all of the additional context: the repo, rev, and file path.

Reply:
Yeah, that was the case I thought of. Seems like that cannot happen. And the other properties pike eventType are not important? I.e. it’s okay here if a click gets filtered out (after the token was hovered)?

Reply:
`eventType` is important downstream, but the downstream caller uses `resolvedPositionEvents`, which also includes the `eventType` property. Consumers of `resolvedPositions` don't need the other properties.